app/views/session.py

```diff
@@ -70,9 +72,13 @@ def login():
     if 'account_service_url' in claims and claims.get('account_service_url'):
         cookie_session[ACCOUNT_URL] = claims.get('account_service_url')
 
-    current_location = get_completeness(current_user).get_latest_location()
+    routing_path = path_finder.get_full_routing_path()
+    completeness = get_completeness(current_user)
+    route = Router(g.schema, routing_path, completeness)
 
-    return redirect(current_location.url(claims))
+    next_location = route.get_next_location()
+
+    return redirect(next_location.url(claims))
```

Comment:
This isn't related to changes in this PR but why do we pass the claims here?

Reply:
Yeh maybe we should refactor this in another PR, we pass in claims but just use these 2 values. we are currently passing `collection_id` into all of the views and only using it for redirects to other views and never actually using it.

Reply:
Tbh, not too sure what was within the scope, I just thought I'll refactor the things that Router can currently cope with and do the rest along with this [card](trello.com/c/R3yLORjb/2244-move-routing-logic-to-a-new-component). 

Regarding claims Im not too sure, its used in multiples places and we just pass it the metadata/claims and it gets the info it needs from it. Something for another card to change that?

Reply:
Oh yeah good catch. That doesn't make sense. Could one of you raise a card to explicitly pass the bits we need to construct the URL?